geometry_msgs/Point p
geometry_msgs/Vector3 g
float32 d
float32 kappa